File: src/lowlevel/client/icefsoperators/icefsSetLk.cpp
/*
 * @Author: Tan90degrees tangentninetydegrees@gmail.com
 * @Date: 2023-03-30 04:19:29
 * @LastEditors: Tan90degrees tangentninetydegrees@gmail.com
 * @LastEditTime: 2023-04-20 11:18:14
 * @FilePath: /icefs/src/lowlevel/client/icefsoperators/icefsSetLk.cpp
 * @Description:
 *
 * Copyright (C) 2023 Tan90degrees <tangentninetydegrees@gmail.com>.
 */
#include <stdio.h>

#include <string>

#include "icefsClient.hpp"
#include "icefsClientUtils.hpp"

void IcefsClient::DoIcefsSetLk(fuse_req_t fuseReq, fuse_ino_t inode,
                               struct fuse_file_info *fi, struct flock *lock,
                               int sleep) {
  ICEFS_PR_FUNCTION;

  fuse_reply_err(fuseReq, ENOTSUP);
}

File: src/lowlevel/server/icefsoperators/icefsSetLk.go
/*
 * @Author: Tan90degrees tangentninetydegrees@gmail.com
 * @Date: 2023-03-11 07:18:32
 * @LastEditors: Tan90degrees tangentninetydegrees@gmail.com
 * @LastEditTime: 2023-04-18 14:39:36
 * @FilePath: /icefs/src/lowlevel/server/icefsoperators/icefsSetLk.go
 * @Description:
 *
 * Copyright (C) 2023 Tan90degrees <tangentninetydegrees@gmail.com>.
 */
package icefsoperators

import (
	"context"
	pb "icefs-server/icefsgrpc"
	"icefs-server/icefsthrift"
	"syscall"
)

func (s *IcefsServer) doIcefsSetLk(fakeInode uint64, sleep int32) (status int32) {
	return int32(syscall.ENOTSUP)
}

func (s *IcefsGRpcServer) DoIcefsSetLk(ctx context.Context, req *pb.IcefsSetLkReq) (*pb.IcefsSetLkRes, error) {
	var res pb.IcefsSetLkRes

	res.Status = s.server.doIcefsSetLk(req.Inode, req.Sleep)

	return &res, nil
}

func (s *IcefsThriftServer) DoIcefsSetLk(ctx context.Context, req *icefsthrift.IcefsSetLkReq) (*icefsthrift.IcefsSetLkRes, error) {
	var res icefsthrift.IcefsSetLkRes

	res.Status = s.server.doIcefsSetLk(uint64(req.Inode), req.Sleep)

	return &res, nil
}
